src/starcheck.pl

```diff
@@ -473,34 +482,55 @@
     push @sim_trans, $_ if ($_->{cmd} eq 'SIMTRANS');
 }
 
+
+sub catalog_array{
+    my $cat = shift;
+    my @catarr;
+    for $i (1 .. 16){
+        if ($cat->{"TYPE$i"} eq 'NUL'){
+            next;
+        }
+        my %catrow = ('yang'=>$cat->{"YANG$i"},
+                      'zang'=>$cat->{"ZANG$i"},
+                      'halfw'=>$cat->{"HALFW$i"},
+                      'type'=>$cat->{"TYPE$i"},
+                      'idx'=>$i);
+        push @catarr, \%catrow;
+    }
+    return \@catarr;
+}
+
+
 # Do main checking
 foreach my $obsid (@obsid_id) {
     $obs{$obsid}->get_agasc_stars($agasc_dir);
     $obs{$obsid}->identify_stars();
-    if ($par{plot}) {
-	eval{
-	    $obs{$obsid}->plot_stars("$STARCHECK/stars_$obs{$obsid}->{obsid}.png") ;
-	    $obs{$obsid}->plot_star_field("$STARCHECK/star_view_$obs{$obsid}->{obsid}.png") ;
-	    $obs{$obsid}->plot_compass("$STARCHECK/compass$obs{$obsid}->{obsid}.png");
-	};
-	if ($@){
-	    print STDERR "Problem with plots for Obsid $obs{$obsid}->{obsid}\n error: $@ will wait 5 sec and retry...\n";
-            sleep(5);
-            # repeat the block using an ugly code copy
-            eval{
-              $obs{$obsid}->plot_stars("$STARCHECK/stars_$obs{$obsid}->{obsid}.png") ;
-              $obs{$obsid}->plot_star_field("$STARCHECK/star_view_$obs{$obsid}->{obsid}.png") ;
-              $obs{$obsid}->plot_compass("$STARCHECK/compass$obs{$obsid}->{obsid}.png");
-            };
-            if ($@){
-              print STDERR " Retry not successful for $obs{$obsid}->{obsid}\n";
-              warning ("Could not create plots for Obsid $obs{$obsid}->{obsid}.\n error $@");
-            }
-            else{
-              print STDERR " Retry successful for $obs{$obsid}->{obsid}\n";
-            }
-          }
-      }
+
+    eval{
+        local $SIG{ALRM}=sub{ die " timed out\n"};
+        eval{
+            alarm 300;
+            my $cat = Ska::Starcheck::Obsid::find_command($obs{$obsid}, "MP_STARCAT");
+            my $cat_as_array = catalog_array($cat);
+            my %plot_args = (obsid=>"$obs{$obsid}->{obsid}",
+                             ra=>$obs{$obsid}->{ra},
+                             dec=>$obs{$obsid}->{dec},
+                             roll=>$obs{$obsid}->{roll},
+                             catalog=>$cat_as_array,
+                             starcat_time=>"$obs{$obsid}->{date}",
+                             outdir=>$STARCHECK);
+            plot_cat_wrapper(\%plot_args);
+            alarm 0;
+        };
+        if ($@){
+            push @global_warn, "ERROR: $@";
+            die "$@\n";
+        }
+        alarm 0;
```

Comment:
I was planning to get rid of the alarms altogether.  I don't see what value they add at this point.  And regarding the Perl checks to see if files are there, you are right and that is both old and an old comment... I don't see much need to check to see if the plots are there if there are no errors  (if I find ways to break Inline::Python in testing where the plotting dies silently, I'll rethink this).